1-starter-app/templates/filter.html

```diff
@@ -0,0 +1,151 @@
+            </div>
+        </div>
+    
+        <div class="row g-3 mb-2 justify-content-center">
+            <div class="col-md-2 offset-md-3">
+                Max date/time:
+            </div>
+            <div class="col-md-2">
+                <input type="date" id="rowKeyDateEndFilter" name="rowKeyDateEnd" class="form-control">
+            </div>
+            <div class="col-md-2">
+                <input type="time" id="rowKeyTimeEndFilter" name="rowKeyTimeEnd" class="form-control">
+            </div>
+        </div>
+    
+        <div class="row g-3 mb-2 justify-content-center">
+            <div class="col-md-3 fw-bold">
+                Temperature
+            </div>
+            <div class="col-md-2">
+                Min temperature:
+            </div>
+            <div class="col-md-4">
+                <input type="text" id="minTemperatureFilter" name="minTemperature" class="form-control" placeholder="Min temperature">
+            </div>
+        </div>
+    
+        <div class="row g-3 mb-2 justify-content-center">
+            <div class="col-md-2 offset-md-3">
+                Max temperature:
+            </div>
+            <div class="col-md-4">
+                <input type="text" id="maxTemperatureFilter" name="maxTemperature" class="form-control" placeholder="Max temperature">
+            </div>
+        </div>
+    
+        <div class="row g-3 mb-2 justify-content-center">
+            <div class="col-md-3 fw-bold">
+                Precipitation
+            </div>
+            <div class="col-md-2">
+                Min precipitation:
+            </div>
+            <div class="col-md-4">
+                <input type="text" id="minPrecipitationFilter" name="minPrecipitation" class="form-control" placeholder="Min precipitation">
+            </div>
+        </div>
+    
+        <div class="row g-3 mb-2 justify-content-center">
+            <div class="col-md-2 offset-md-3">
+                Max precipitation:
+            </div>
+            <div class="col-md-4">
+                <input type="text" id="maxPrecipitationFilter" name="maxPrecipitation" class="form-control" placeholder="Max precipitation">
+            </div>
+        </div>
+    
+        <div class="row g-3 mb-2 justify-content-center">
+            <div class="col-md-4 offset-md-5 text-end">
+                <button id="submit" class="btn btn-success" onclick="javascript:findByFilter()">Filter data</button>
+            </div>
+        </div>
+    </div>
+    <hr>
+    <h3>Weather Data</h3>
+    <table class="table table-sm mb-1" id="result-table">
+    </table>
+    
+    <footer class="border-top footer text-muted">
+        <div class="container">
+            &copy; 2021 - AzureTablesDemoApplication - <a id="privacyLink">Privacy</a>
+        </div>
+    </footer>
+    <script type="text/javascript" src="./lib/jquery/dist/jquery.js"></script>
+    <script type="text/javascript" src="./lib/bootstrap/dist/js/bootstrap.js"></script>
```

Comment:
I think we should reference CDN resources for these dependencies instead of hosting them in the project. for example

```html
<script src="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/js/bootstrap.bundle.min.js" integrity="sha384-ka7Sk0Gln4gmtz2MlQnikT1wXgYsOg+OMhuP+IlRH9sENBO0LRn5q+8nbTov4+1p" crossorigin="anonymous"></script>
```